src/wallets/injected/InjectedMathWallet.ts

```diff
@@ -0,0 +1,44 @@
+import { PublicKey, Signature } from "near-api-js/lib/utils/key_pair";
+
+interface MathAccount {
+  networkUnique: string;
+  address: string;
+  name: string;
+  authority: string;
+  selected: boolean;
+  type: string;
+
+  accountId: string;
+  publicKey: string;
+  permission: string;
+  network: string;
+}
+
+interface MathNetwork {
+  id: string;
+  name: string;
+  blockchain: string;
+  chainId: string;
+  nodeAddress: string;
+  nodeAddress2: string;
+  nodeAddress3: string;
+  enable: boolean;
+  custom: boolean;
+  extra: string;
+}
+
+export interface InjectedMathWallet {
+  signer: {
+    account: MathAccount;
+    network: MathNetwork;
+    createKey(accountId: string, networkId?: string): Promise<PublicKey>;
+    getPublicKey(accountId?: string, networkId?: string): Promise<PublicKey>;
```

Comment:
You're right account can be null too.

As for the `createKey`, `getPublicKey`, `signMessage` methods I added them only because typescript for `line 152 in MathWallets.ts` was complaining:

`Type '{ account: MathAccount | null; network: MathNetwork; }' is missing the following properties from type 'Signer': createKey, getPublicKey, signMessage`

Reply:
Sorry, I was specifically talking about whether the parameters for these methods should have the `?` (optional) flag?

Sounds like you've noted `getPublicKey` can return `null`

Reply:
Oh those are the definitions I got from the near-api-js library:
https://github.com/near/near-api-js/blob/master/src/signer.ts